nixos/modules/services/web-apps/photoprism.nix

```diff
@@ -0,0 +1,291 @@
+      '';
+    };
+
+    host = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      example = "0.0.0.0";
+      description = "The address to serve the web interface at.";
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 2342;
+      description = "The port to serve the web interface at.";
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Open the web interface port in the firewall for photoprism.";
+    };
+
+    databaseDriver = mkOption {
+      type = types.enum [ "sqlite" "mysql" ];
+      default = "sqlite";
+      description = ''
+        Database driver, select <literal>"sqlite"</literal> to use an embedded database.
+
+        Photoprism recommends a MariaDB database (using the
+        <literal>"mysql"</literal> driver) to improve performance.
+      '';
+    };
+
+    databaseDsn = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = ''
+        photoprism@unix(/run/mysqld/mysqld.sock)/photoprism?charset=utf8mb4,utf8&parseTime=true
+      '';
+      description = ''
+        Photoprism database source name. Valid only when
+        <link linkend="opt-services.photoprism.databaseDriver">databaseDriver</link>
+        is set to <literal>"mysql"</literal>.
+      '';
+    };
+
+    initialPassword = mkOption {
+      type = types.str;
+      default = "insecure";
+      description = ''
+        Initial admin password.
+
+        This should be changed in the web UI after the first startup.
+      '';
+    };
+
+    detectNsfw = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Flag photos as private that MAY be offensive.
+      '';
+    };
+
+    allowNsfw = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Allow uploads that MAY be offensive.
+      '';
+    };
+
+    experimental = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable experimental features.
+      '';
+    };
+
+    originalsLimit = mkOption {
+      type = types.int;
+      default = 5000;
+      description = ''
+        File size limit for originals in MB (increase for high-res video).
+      '';
+    };
+
+    public = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        No authentication required (disables password protection).
+      '';
+    };
+
+    readonly = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Do not modify originals directory (reduced functionality).
+      '';
+    };
+
+    originalsDir = mkOption {
+      type = types.path;
+      example = "/mnt/nas/photos";
+      description = "Path to your photos.";
+    };
+
+    dataDir = mkOption {
+      type = types.path;
+      default = "/var/lib/photoprism";
```

Comment:
Let's not make this configurable.

Reply:
Photoprism stores a lot of data. Including resized and edited images. I think we should allow users to, for example, store them in `/srv` if they want to. Most modules do AFAIK.

Reply:
Modules used to do this but now we prefer `StateDirectory` because it works well with `DynamicUser`.

Reply:
Makes sense! Thanks for clarifying